tests/test_execution_compute_plan.py

```diff
@@ -303,15 +303,17 @@ def test_compute_plan_aggregate_composite_traintuples(global_execution_env):
     """
     Compute plan version of the `test_aggregate_composite_traintuples` method from `test_execution.py`
     """
-    factory, network = global_execution_env
+    factory, state, network = global_execution_env
     sessions = [s.copy() for s in network.sessions]
 
     aggregate_worker = sessions[0].node_id
     number_of_rounds = 2
 
     # register objectives, datasets, and data samples
-    datasets = sessions[0].state.datasets + sessions[1].state.datasets
-    objectives = sessions[0].state.objectives[:0] + sessions[1].state.objectives[:0]
+    datasets = [d for d in state.datasets if d.owner == sessions[0].node_id] + \
+               [d for d in state.datasets if d.owner == sessions[1].node_id]
+    objectives = [o for o in state.objectives if o.owner == sessions[0].node_id][:0] + \
+                 [o for o in state.objectives if o.owner == sessions[1].node_id][:0]
```

Comment:
I think this could be simplified by taking all datasets and objectives in the state